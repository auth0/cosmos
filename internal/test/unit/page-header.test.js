import React from 'react'
import { shallow, mount } from 'enzyme'

import { PageHeader, Button, Icon, Link } from '@auth0/cosmos'
import { escapeComponent } from 'uri-js'

describe('Page Header', () => {
  it('renders primary action only', () => {
    const label = 'Button Content'
    const icon = 'copy'
    const handler = jest.fn()
    const wrapper = shallow(<PageHeader primaryAction={{ label, icon, handler }} />)

    const headerButtons = wrapper.find(Button)
    const primaryAction = headerButtons.first()
    primaryAction.simulate('click')

    expect(headerButtons).toHaveLength(1)
    expect(handler).toHaveBeenCalled()
    expect(wrapper).toMatchSnapshot()
  })

  it('renders secondary action only', () => {
    const label = 'Button Content'
    const icon = 'copy'
    const handler = jest.fn()
    const wrapper = shallow(
      <PageHeader description="A description" secondaryAction={{ label, icon, handler }} />
    )

    const headerButtons = wrapper.find(Button)
    const secondaryAction = headerButtons.last()
    secondaryAction.simulate('click')

    expect(headerButtons).toHaveLength(1)
    expect(handler).toHaveBeenCalled()
    expect(wrapper).toMatchSnapshot()
  })

  it('renders both primary and secondary actions', () => {
    const actions = {
      primary: {
        label: 'Primary',
        icon: 'analytics',
        handler: jest.fn()
      },
      secondary: {
        label: 'Secondary',
        icon: 'copy',
        handler: jest.fn()
      }
    }

    const wrapper = shallow(
      <PageHeader
        primaryAction={actions.primary}
        secondaryAction={actions.secondary}
        description="A description"
      />
    )

    const headerButtons = wrapper.find(Button)
    const primaryAction = headerButtons.first()
    const secondaryAction = headerButtons.last()

    primaryAction.simulate('click')
    secondaryAction.simulate('click')

    expect(headerButtons).toHaveLength(2)

    expect(actions.primary.handler).toHaveBeenCalled()
    expect(actions.secondary.handler).toHaveBeenCalled()
    expect(wrapper).toMatchSnapshot()
  })

  it('calls primaryAction#handler callback', () => {
    const primaryAction = { label: 'Something', icon: 'copy', handler: jest.fn() }
    const pageHeader = shallow(
      <PageHeader primaryAction={primaryAction} description="A description" />
    )
    pageHeader
      .childAt(2)
      .childAt(0)
      .simulate('click')

    expect(primaryAction.handler).toHaveBeenCalled()
  })

  it('calls secondaryAction#handler callback', () => {
    const secondaryAction = { label: 'Something', icon: 'copy', handler: jest.fn() }
    const pageHeader = shallow(
      <PageHeader secondaryAction={secondaryAction} description="A description" />
    )
    pageHeader
      .childAt(2)
      .childAt(0)
      .simulate('click')

    expect(secondaryAction.handler).toHaveBeenCalled()
  })
})
