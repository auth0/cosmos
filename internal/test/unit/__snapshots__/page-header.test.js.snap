// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Header renders both primary and secondary actions 1`] = `
<styled.div
  data-cosmos-key="page-header"
>
  <Heading
    size={1}
  />
  <Component
    description="A description"
    primaryAction={
      Object {
        "handler": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "icon": "analytics",
        "label": "Primary",
      }
    }
    secondaryAction={
      Object {
        "handler": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "icon": "copy",
        "label": "Secondary",
      }
    }
    title=""
  />
  <ButtonGroup
    align="left"
    compressed={false}
  >
    <Button
      appearance="secondary"
      disabled={false}
      icon="copy"
      iconAlign="left"
      loading={false}
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      size="large"
      success={false}
    >
      Secondary
    </Button>
    <Button
      appearance="cta"
      disabled={false}
      icon="analytics"
      iconAlign="left"
      loading={false}
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      size="large"
      success={false}
    >
      Primary
    </Button>
  </ButtonGroup>
</styled.div>
`;

exports[`Page Header renders primary action only 1`] = `
<styled.div
  data-cosmos-key="page-header"
>
  <Heading
    size={1}
  />
  <Component
    primaryAction={
      Object {
        "handler": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "icon": "copy",
        "label": "Button Content",
      }
    }
    title=""
  />
  <ButtonGroup
    align="left"
    compressed={false}
  >
    <Button
      appearance="cta"
      disabled={false}
      icon="copy"
      iconAlign="left"
      loading={false}
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      size="large"
      success={false}
    >
      Button Content
    </Button>
  </ButtonGroup>
</styled.div>
`;

exports[`Page Header renders secondary action only 1`] = `
<styled.div
  data-cosmos-key="page-header"
>
  <Heading
    size={1}
  />
  <Component
    description="A description"
    secondaryAction={
      Object {
        "handler": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "icon": "copy",
        "label": "Button Content",
      }
    }
    title=""
  />
  <ButtonGroup
    align="left"
    compressed={false}
  >
    <Button
      appearance="secondary"
      disabled={false}
      icon="copy"
      iconAlign="left"
      loading={false}
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      size="large"
      success={false}
    >
      Button Content
    </Button>
  </ButtonGroup>
</styled.div>
`;
